pandas/io/parquet.py

```diff
@@ -136,6 +136,10 @@ def validate_dataframe(df: DataFrame):
         if not valid_names:
             raise ValueError("Index level names must be strings")
 
+        # Paquet format does not support saving float16
```

Comment:
well does this actually fail in pyarrow or fastparquet?

we barely support float16 in pandas anyhow

Reply:
It fails in PyArrow, Parquet format does not support saving fp16 files but fastparquet seems to typecast fp16 to fp32 before saving.

Release Note: 'Added validation checks for float16 dtype in pandas.DataFrame.to_parquet(engine="pyarrow")'